_posts/2015-05-30-the-state-of-javascript.md

```diff
@@ -0,0 +1,122 @@
+if you'd like to discuss things further.
+
+# Predictions
+
+I made 8 predictions on what I think we'll see in the next 12 months, and most
+of these are influenced by the three core goals of ECMAScript 2015, which hopes
+to provide a better language for:
+
+- complex applications
+- libraries
+- code generation (languages that compile to JS)
+
+### 1: Fewer people will write JavaScript without a compilation step
+
+We're seeing this trend already, libraries like
+[TypeScript](http://www.typescriptlang.org/) and [Babel](http://babeljs.io/)
+have built on what [CoffeeScript](http://coffeescript.org/) showed people
+wanted, by building on top of JavaScript and compiling down to JavaScript.
+CoffeeScript deserves a lot of credit here, it was the first project that really
+did this and showed that people were willing to trade a slightly more complex
+workflow for additional functionality.
+
+Babel is slightly different in that all the new functionality it provides is
+part of ECMAScript 2015 (ES6) or beyond, so everything it implements in theory
+will eventually be in the browser. Going forward, Babel, TypeScript and
+[ClojureScript](https://github.com/clojure/clojurescript) will probably be the
+three I'd back to become even more popular.
+
+### 2: Smaller libraries (and the composing of) will be preferred over large frameworks
+
+Alongside the larger, fully featured frameworks of Angular, Ember and others,
+there's a myriad of smaller libraries that focus on doing one thing, and doing
+it really well. You could even argue that ReactJS is a good example of this; as
+a library it provides just the view layer for an application, and nothing more.
+Given that npm provides a (relatively) easy way to install and manage all these
+libraries, I think it will become more common for developers to construct their
+own stacks of small libraries that can be swapped in and out, over using a large
+framework where you're stuck with what it provides.
+
+### 3: Focus on libraries that do one thing and one thing well
+
+Related very much to the previous point, I think that there will be an even
+bigger focus on the development and release of libraries that exist to solve one
+problem, and do it very well. This feels like a natural process as an ecosystem
+matures, and we figure out the best solutions to new problems (such as client
+side "MVC" approaches). Write write an entire framework when you could write a
+small library to plug the one problem you need to fix, and then couple it with
+some other libraries that provide the rest of the functionality you need.
+
+### 4: Large, fully featured frameworks will remain rightly popular
+
+The previous two thoughts might make you think that I'm predicting the demise of
+Angular, Ember and so on. This is definitely not the case. There will always be
+(and quite rightly so) a use case and need for this larger frameworks. I expect
+their popularity to grow and them to keep improving over time as they have been
```

Comment:
this last line could be reworded, or even removed? That they'll improve is hopefully a given. If they don't, they'll die. And why do you think their popularity will grow? It sounds as if you're suggesting people will continue using big frameworks, and at the same time an increasing number of people will compose libraries; wouldn't that mean that framework usage is likely to stay the same, or even reduce?